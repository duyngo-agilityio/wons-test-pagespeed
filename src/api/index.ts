export * from './invoice';
export * from './product';
export * from './statistic';
export * from './user';
export * from './customer';
export * from './task';
export * from './calendar';
export * from './image';
