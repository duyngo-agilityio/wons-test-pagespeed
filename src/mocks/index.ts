export * from './customer';
export * from './recentServices';
export * from './tabs';
