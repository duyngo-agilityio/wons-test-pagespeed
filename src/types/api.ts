export enum Method {
  Get = 'GET',
  Post = 'POST',
  Put = 'PUT',
  Patch = 'PATCH',
  Delete = 'DELETE',
}
