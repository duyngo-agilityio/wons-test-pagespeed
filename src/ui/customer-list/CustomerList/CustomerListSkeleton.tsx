'use client';

import { Skeleton } from '@nextui-org/react';

// Components
import { TableSkeleton } from '@/components';

export const MAPPING_CUSTOMER_LIST_SKELETON = [
  {
    header: 'Name',
    accessor: () => (
      <div className="flex gap-3.5 items-center min-w-36">
        <Skeleton className="h-9 w-9 rounded-full" />
        <Skeleton className="h-8 w-3/5 rounded-lg" />
      </div>
    ),
  },
  {
    header: 'Email',
    accessor: () => <Skeleton className="h-8 w-4/5 rounded-lg" />,
  },
  {
    header: 'Phone Number ',
    accessor: () => <Skeleton className="h-8 w-3/5 rounded-lg" />,
  },
  {
    header: 'Gender',
    accessor: () => <Skeleton className="h-8 w-3/5 rounded-lg" />,
  },
  {
    accessor: () => <Skeleton className="h-8 w-3/5 rounded-lg" />,
  },
];

const CustomerListSkeleton = () => (
  <TableSkeleton
    variant="primary"
    isStriped={false}
    columns={MAPPING_CUSTOMER_LIST_SKELETON}
  />
);

export default CustomerListSkeleton;
