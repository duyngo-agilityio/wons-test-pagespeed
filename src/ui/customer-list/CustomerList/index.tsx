// apis
import { getCustomers } from '@/api';

// uis
import { CustomerListClient } from '@/ui';

type TCustomerListProps = {
  page?: number;
};

const CustomerList = async ({ page }: TCustomerListProps) => {
  const { data: customerRes, meta } = await getCustomers({ page });

  const { pagination } = meta || {};
  const { pageCount = 0 } = pagination || {};

  return (
    <CustomerListClient customerList={customerRes} pageCount={pageCount} />
  );
};

export default CustomerList;
