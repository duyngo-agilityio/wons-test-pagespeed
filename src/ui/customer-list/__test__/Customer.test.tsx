import { render, waitFor } from '@testing-library/react';
import { FunctionComponent } from 'react';

// components
import CustomerList from '../CustomerList';

// mocks
import { CUSTOMER_MOCK } from '@/mocks';

// apis
import { getCustomers } from '@/api';

jest.mock('@/api', () => ({
  ...jest.requireActual('@/api'),
  getCustomers: jest.fn(),
}));

async function resolvedComponent<T>(Component: FunctionComponent<T>, props: T) {
  const ComponentResolved = await Component(props);
  return () => ComponentResolved;
}

beforeAll(() => {
  global.fetch = jest.fn(() =>
    Promise.resolve({
      ok: true,
      status: 200,
      json: () => Promise.resolve({}),
      headers: new Headers({
        'Content-Type': 'application/json',
      }),
      redirected: false,
      statusText: 'OK',
      type: 'basic',
      url: '',
      clone: jest.fn(),
      body: null,
      bodyUsed: false,
      arrayBuffer: jest.fn(),
      blob: jest.fn(),
      formData: jest.fn(),
      text: jest.fn(),
    } as Response),
  );
});

afterAll(() => {
  global.fetch = originalFetch;
});

const originalFetch = global.fetch;

describe('Customer List component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render all content with resolved data and match snapshot', async () => {
    (getCustomers as jest.Mock).mockResolvedValue({
      data: CUSTOMER_MOCK,
    });
    const CustomerListResolved = await resolvedComponent(CustomerList, {});
    const { container } = render(<CustomerListResolved />);

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
