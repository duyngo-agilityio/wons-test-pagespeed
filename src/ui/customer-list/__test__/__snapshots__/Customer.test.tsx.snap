// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer List component should render all content with resolved data and match snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-10"
  >
    <div
      class="flex-col relative gap-4 w-full grid"
    >
      <div
        class="z-0 flex flex-col relative justify-between gap-4 bg-content1 overflow-auto rounded-large w-full p-0 shadow-none dark:bg-gray-400"
      >
        <table
          aria-describedby=""
          class="min-w-full h-auto table-auto w-full border-separate border-spacing-x-0 border-spacing-y-[10px] bg-gray-50 dark:bg-gray-600"
          id="react-aria-:r0:"
          role="grid"
          tabindex="0"
        >
          <thead
            class="[&>tr]:first:rounded-lg"
            role="rowgroup"
          >
            <tr
              class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              role="row"
            >
              <th
                class="group h-10 align-middle whitespace-nowrap text-foreground-500 text-tiny first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start first:rounded-l-none last:rounded-r-none py-0 cursor-pointer font-normal bg-gray-50 dark:bg-gray-600 px-[17px]"
                data-key="Name0"
                data-testid="sort-btn"
                id="react-aria-:r0:-Name0"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="flex items-center gap-2"
                >
                  Name
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m98 190.06 139.78 163.12a24 24 0 0 0 36.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
                    />
                  </svg>
                </div>
              </th>
              <th
                class="group h-10 align-middle whitespace-nowrap text-foreground-500 text-tiny first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start first:rounded-l-none last:rounded-r-none py-0 cursor-pointer font-normal bg-gray-50 dark:bg-gray-600 px-[17px]"
                data-key="Email1"
                data-testid="sort-btn"
                id="react-aria-:r0:-Email1"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="flex items-center gap-2"
                >
                  Email
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m98 190.06 139.78 163.12a24 24 0 0 0 36.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
                    />
                  </svg>
                </div>
              </th>
              <th
                class="group h-10 align-middle whitespace-nowrap text-foreground-500 text-tiny first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start first:rounded-l-none last:rounded-r-none py-0 cursor-pointer font-normal bg-gray-50 dark:bg-gray-600 px-[17px]"
                data-key="Phone Number2"
                data-testid="sort-btn"
                id="react-aria-:r0:-PhoneNumber2"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="flex items-center gap-2"
                >
                  Phone Number
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m98 190.06 139.78 163.12a24 24 0 0 0 36.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
                    />
                  </svg>
                </div>
              </th>
              <th
                class="group h-10 align-middle whitespace-nowrap text-foreground-500 text-tiny first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start first:rounded-l-none last:rounded-r-none py-0 cursor-pointer font-normal bg-gray-50 dark:bg-gray-600 px-[17px]"
                data-key="Gender3"
                data-testid="sort-btn"
                id="react-aria-:r0:-Gender3"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="flex items-center gap-2"
                >
                  Gender
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m98 190.06 139.78 163.12a24 24 0 0 0 36.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
                    />
                  </svg>
                </div>
              </th>
            </tr>
            <tr
              aria-hidden="true"
              class="w-px h-px block"
              style="margin-left: 0.25rem; margin-top: 0.25rem;"
              tabindex="-1"
            />
          </thead>
          <tbody
            role="rowgroup"
          >
            <tr
              aria-labelledby="react-aria-:r0:-1-Name0"
              class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 bg-white dark:bg-gray-400"
              data-first="true"
              data-id="1"
              data-key="1"
              role="row"
              tabindex="-1"
            >
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-0"
                id="react-aria-:r0:-1-Name0"
                role="rowheader"
                tabindex="-1"
              >
                <div
                  class="flex gap-3.5 items-center"
                >
                  <div
                    class="relative w-9 h-9 rounded-full"
                  >
                    <img
                      alt="customer avatar"
                      class="rounded-full "
                      data-nimg="fill"
                      decoding="async"
                      src="http://localhost/"
                      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                    />
                  </div>
                  <p
                    class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                  >
                     
                  </p>
                </div>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-1"
                role="gridcell"
                tabindex="-1"
              >
                <div
                  class="flex gap-2.5 items-center"
                >
                  <p
                    class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                  />
                </div>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-2"
                role="gridcell"
                tabindex="-1"
              >
                <p
                  class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                />
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-3"
                role="gridcell"
                tabindex="-1"
              >
                <div
                  class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap h-7 text-small text-default-foreground w-[100px] px-2 rounded-[20px] bg-teal-500/10 dark:bg-green-200"
                >
                  <span
                    class="flex-1 font-normal px-2 text-center text-md capitalize text-teal-500 dark:text-teal-300"
                  >
                    male
                  </span>
                </div>
              </td>
            </tr>
            <tr
              aria-labelledby="react-aria-:r0:-2-Name0"
              class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 bg-white dark:bg-gray-400"
              data-id="2"
              data-key="2"
              data-middle="true"
              data-odd="true"
              role="row"
              tabindex="-1"
            >
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-0"
                id="react-aria-:r0:-2-Name0"
                role="rowheader"
                tabindex="-1"
              >
                <div
                  class="flex gap-3.5 items-center"
                >
                  <div
                    class="relative w-9 h-9 rounded-full"
                  >
                    <img
                      alt="customer avatar"
                      class="rounded-full "
                      data-nimg="fill"
                      decoding="async"
                      src="http://localhost/"
                      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                    />
                  </div>
                  <p
                    class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                  >
                     
                  </p>
                </div>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-1"
                role="gridcell"
                tabindex="-1"
              >
                <div
                  class="flex gap-2.5 items-center"
                >
                  <p
                    class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                  />
                </div>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-2"
                role="gridcell"
                tabindex="-1"
              >
                <p
                  class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                />
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-3"
                role="gridcell"
                tabindex="-1"
              >
                <div
                  class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap h-7 text-small text-default-foreground w-[100px] px-2 rounded-[20px] bg-teal-500/10 dark:bg-green-200"
                >
                  <span
                    class="flex-1 font-normal px-2 text-center text-md capitalize text-teal-500 dark:text-teal-300"
                  >
                    male
                  </span>
                </div>
              </td>
            </tr>
            <tr
              aria-labelledby="react-aria-:r0:-3-Name0"
              class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 bg-white dark:bg-gray-400"
              data-id="3"
              data-key="3"
              data-last="true"
              role="row"
              tabindex="-1"
            >
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-0"
                id="react-aria-:r0:-3-Name0"
                role="rowheader"
                tabindex="-1"
              >
                <div
                  class="flex gap-3.5 items-center"
                >
                  <div
                    class="relative w-9 h-9 rounded-full"
                  >
                    <img
                      alt="customer avatar"
                      class="rounded-full "
                      data-nimg="fill"
                      decoding="async"
                      src="http://localhost/"
                      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                    />
                  </div>
                  <p
                    class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                  >
                     
                  </p>
                </div>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-1"
                role="gridcell"
                tabindex="-1"
              >
                <div
                  class="flex gap-2.5 items-center"
                >
                  <p
                    class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                  />
                </div>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-2"
                role="gridcell"
                tabindex="-1"
              >
                <p
                  class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                />
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-3"
                role="gridcell"
                tabindex="-1"
              >
                <div
                  class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap h-7 text-small text-default-foreground w-[100px] px-2 rounded-[20px] bg-teal-500/10 dark:bg-green-200"
                >
                  <span
                    class="flex-1 font-normal px-2 text-center text-md capitalize text-teal-500 dark:text-teal-300"
                  >
                    male
                  </span>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    class="EZDrawer"
    id="EZDraweranzg4"
  >
    <input
      class="EZDrawer__checkbox"
      id="EZDrawer__checkboxanzg4"
      type="checkbox"
    />
    <nav
      class="EZDrawer__container !w-[302px] !max-w-[302px]"
      id="EZDrawer__containeranzg4"
      role="navigation"
      style="z-index: 101; transition-duration: 500ms; top: 0px; right: 0px; transform: translate3d(100%, 0, 0); width: 250px; height: 100vh;"
    >
      <div
        class="min-h-full max-w-[302px] bg-white dark:bg-gray-400 py-[62px] px-6"
      >
        <div
          class="flex flex-col items-center gap-4 border-b-1 border-blue-800/10 dark:border-white/10 pb-7.5"
        >
          <div
            class="relative w-[70px] h-[70px]"
          >
            <img
              alt="Jane-Smith"
              class="object-contain rounded-full"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="http://localhost/_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
            />
          </div>
          <div
            class="flex flex-col items-center"
          >
            <p
              class="font-dm-sans text-blue-800 dark:text-white text-3xl font-bold"
            >
              Jane Smith
            </p>
            <p
              class="font-dm-sans text-blue-800 dark:text-white text-md "
            >
              Product Manager
            </p>
          </div>
        </div>
        <div
          class="flex flex-col gap-2.5 mt-10"
        >
          <p
            class="font-dm-sans text-blue-800 dark:text-white text-xxl capitalize font-medium"
          >
            Contact info
          </p>
          <div>
            <div
              class="flex items-center gap-2 border-b-1 border-blue-800/10 dark:border-white/10 py-4"
            >
              <svg
                class="text-blue-800/30 dark:text-white/30"
                color="#3A36DB"
                font-size="16"
                height="14"
                viewBox="0 0 16 14"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.6192 0C12.6622 0 13.6656 0.412222 14.4037 1.15189C15.1426 1.89 15.5556 2.88556 15.5556 3.92778V10.0722C15.5556 12.2422 13.79 14 11.6192 14H3.93556C1.76478 14 0 12.2422 0 10.0722V3.92778C0 1.75778 1.757 0 3.93556 0H11.6192ZM12.8567 5.08673L12.9189 5.02451C13.1048 4.79895 13.1048 4.47229 12.9103 4.24673C12.8022 4.13084 12.6537 4.06007 12.4989 4.04451C12.3356 4.03595 12.18 4.09118 12.0626 4.20007L8.55556 7.00007C8.10444 7.37418 7.45811 7.37418 7 7.00007L3.5 4.20007C3.25811 4.02118 2.92367 4.04451 2.72222 4.25451C2.51222 4.46451 2.48889 4.79895 2.667 5.03229L2.76889 5.1334L6.30778 7.89451C6.74333 8.23673 7.27144 8.4234 7.82444 8.4234C8.37589 8.4234 8.91333 8.23673 9.34811 7.89451L12.8567 5.08673Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <p
                class="font-dm-sans text-blue-800 dark:text-white text-xl opacity-70 font-medium"
              >
                jane.smith@example.com
              </p>
            </div>
            <div
              class="flex items-center gap-2 border-b-1 border-blue-800/10 dark:border-white/10 py-4"
            >
              <svg
                class="text-blue-800/30 dark:text-white/30"
                fill="currentColor"
                font-size="16"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                version="1.1"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 10c-1 1-1 2-2 2s-2-1-3-2-2-2-2-3 1-1 2-2-2-4-3-4-3 3-3 3c0 2 2.055 6.055 4 8s6 4 8 4c0 0 3-2 3-3s-3-4-4-3z"
                />
              </svg>
              <p
                class="font-dm-sans text-blue-800 dark:text-white text-xl opacity-70 font-medium"
              >
                987-654-3210
              </p>
            </div>
            <div
              class="flex items-center gap-2 py-4"
            >
              <svg
                class="text-blue-800/30 dark:text-white/30"
                fill="currentColor"
                font-size="16"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 384 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M215.7 499.2C267 435 384 279.4 384 192C384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2c12.3 15.3 35.1 15.3 47.4 0zM192 128a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"
                />
              </svg>
              <p
                class="font-dm-sans text-blue-800 dark:text-white text-xl opacity-70 font-medium"
              >
                456 Market St, Anytown, USA
              </p>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <label
      class="EZDrawer__overlay "
      for="EZDrawer__checkboxanzg4"
      id="EZDrawer__overlayanzg4"
      style="background-color: rgb(0, 0, 0); opacity: 0.4; z-index: 100;"
    />
  </div>
</div>
`;
